x/upgrade/internal/keeper/keeper.go

```diff
@@ -4,25 +4,28 @@ import (
 	"encoding/binary"
 	"fmt"
 
+	"github.com/tendermint/tendermint/libs/log"
+
 	"github.com/cosmos/cosmos-sdk/codec"
 	"github.com/cosmos/cosmos-sdk/store/prefix"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/x/upgrade/internal/types"
-	"github.com/tendermint/tendermint/libs/log"
 )
 
 type Keeper struct {
-	storeKey        sdk.StoreKey
-	cdc             *codec.Codec
-	upgradeHandlers map[string]types.UpgradeHandler
+	skipUpgradeHeights []int64
+	storeKey           sdk.StoreKey
+	cdc                *codec.Codec
+	upgradeHandlers    map[string]types.UpgradeHandler
 }
 
 // NewKeeper constructs an upgrade Keeper
-func NewKeeper(storeKey sdk.StoreKey, cdc *codec.Codec) Keeper {
+func NewKeeper(skipUpgradeHeights []int64, storeKey sdk.StoreKey, cdc *codec.Codec) Keeper {
 	return Keeper{
-		storeKey:        storeKey,
-		cdc:             cdc,
-		upgradeHandlers: map[string]types.UpgradeHandler{},
+		skipUpgradeHeights: skipUpgradeHeights,
```

Comment:
If you convert `skipUpgradeHeights` to a map here, you can do a much simpler and quicker lookup than having to implement and use auxiliary functions.

Reply:
Thanks for your suggestion, it made it cleaner